include/nbl/builtin/hlsl/sampling/cos_weighted.hlsl

```diff
@@ -70,16 +70,16 @@ struct ProjectedSphere
         return T(0.5) * hemisphere_t::pdf(L_z);
     }
 
-    template<typename SpectralBins>
-    static bxdf::quotient_and_pdf<SpectralBins, T> quotient_and_pdf(T L)
+    template<typename U>
+    static sampling::quotient_and_pdf<U, T> quotient_and_pdf(T L)
     {
-        return bxdf::quotient_and_pdf<SpectralBins, T>::create(hlsl::promote<SpectralBins>(1.0), pdf(L));
+        return sampling::quotient_and_pdf<U, T>::create(hlsl::promote<U>(1.0), pdf(L));
     }
 
-    template<typename SpectralBins>
-    static bxdf::quotient_and_pdf<SpectralBins, T> quotient_and_pdf(vector_t3 L)
+    template<typename U>
+    static sampling::quotient_and_pdf<U, T> quotient_and_pdf(vector_t3 L)
     {
-        return quotient_and_pdf<SpectralBins>(L.z);
+        return sampling::quotient_and_pdf<U, T>::create(hlsl::promote<U>(1.0), pdf(L.z));
```

Comment:
1. you're already in `sampling` namespace
2. `U` should be required to be a scalar via `enable_if_t` or `NBL_FUNC_REQUIRES` - so no need for `promote` just a `U` wrapper or `_static_cast`

Reply:
The compiler (or msvc at least) confuses the type name with the function name, seemingly because of the overload